src/Cake.Issues.DupFinder/DupFinderIssuesProvider.cs

```diff
@@ -180,7 +212,23 @@ private static bool TryGetLine(XContainer fragment, out int lineStart, out int l
             return true;
         }
 
-        private IReadOnlyCollection<DuplicateFragment> GetDuplicateFragements(XContainer duplicate)
+        private Uri ResolveFileLinkForFragment(DuplicateFragment fragment)
+        {
+            if (string.IsNullOrWhiteSpace(this.Settings?.FileLinkSettings?.FileLinkPattern))
```

Comment:
It should be enough to check that `this.Settings.FileLinkSettings` is not null. There are already checks in place for checking that `this.Settings` is not `null` and `FileLinkPattern` is required to be passed to the ctor of `FileLinkSettings`.